Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverOnboarding/Status.hs

```diff
@@ -379,15 +379,17 @@ fetchInprogressVehicleDocuments personId merchantOpCity transporterConfig langua
         Right registrationNo -> do
           rcNoEnc <- encrypt registrationNo
           rc <- RCQuery.findByCertificateNumberHash (rcNoEnc & hash)
-          isUnlinked <- case rc of
-            Just rc_ -> DRAQuery.findUnlinkedRC personId rc_.id
-            Nothing -> pure []
+          (isUnlinked, mbRcId) <- case rc of
+            Just rc_ -> do
+              iu <- DRAQuery.findUnlinkedRC personId rc_.id
+              pure (iu, Just rc_.id)
+            Nothing -> pure ([], Nothing)
```

Comment:
Are you sure this is correct?
Line 392 will work only when  `not (null isUnlinked) = False`, or `null isUnlinked`, but you put `Nothing` for this case